#Bug, No check on open file option to save changes.

laiqujan opened the pull request:
Fixed.
Hi,
First of all, I really appreciate your work towards open source community.
I really wants to contribute in this project So, I started to test Text-editor and found some bug that there is no check on open file option to save changes if there is text in notepad.
can you please merge my pull request or  add fallowing lines of code in your UI.java  class for open file option.
// If the source was the "open" option

laiqujan committed:
#Bug, No check on open file option to save changes.
Fixed

pH-7 commented:
Hey @laiqujan Thanks very much for your PR. I really appreciate it!
I see that there is a conflict there with some newer changes done in `src/simplejavatexteditor/UI.java` Can you fix it?

pH-7 commented on .gitignore:
```diff
@@ -6,3 +6,4 @@
 # Mac OS System
 .DS_Store*
 ._*
+/bin/
```
I think, we don't really need this. It depends on your own environment, but no need to commit it.
If you can revert it, that would be great.

pH-7 commented on src/simplejavatexteditor/UI.java:
```diff
@@ -472,7 +472,7 @@ public void actionPerformed(ActionEvent e) {
             if (edit) {
                 Object[] options = {"Save", "No Save", "Return"};
-                int n = JOptionPane.showOptionDialog(this, "Do you want to save the file at first ?", "Question",
+                int n = JOptionPane.showOptionDialog(this, "Do you want to save changes ?", "Question",
```
Maybe `Do you want to save the changes?` instead (and remove also the space before the question mark). I know that this isn't you (my bad :)), but since you modified the text, you can do this too.

pH-7 commented on src/simplejavatexteditor/UI.java:
```diff
@@ -487,30 +487,23 @@ else if (e.getSource() == newFile || e.getSource() == newButton) {
-                }
-            }
+        	
+        	  if (edit) {
```
I'm sure you like DRY principle. However, you duplicated the same code as above https://github.com/pH-7/Simple-Java-Text-Editor/pull/27/files#diff-444b0d2f64a93f3dbc07fc4e0eb73f6dR473

Could you avoid doing this. Same for `Object[] options = {"Save", "No Save", "Return"};` array, `JOptionPane.YES_NO_CANCEL_OPTION, ...`, ..., etc.

This block should be refactoring instead.

pH-7 replied:
I again @laiqujan I think you didn't read my comment or forgot it :)

pH-7 replied:
I again @laiqujan I think you didn't read my comment or forgot it :)

pH-7 replied:
I again @laiqujan I think you didn't read my comment or forgot it :)

pH-7 replied:
I again @laiqujan I think you didn't read my comment or forgot it :)

pH-7 replied:
I again @laiqujan I think you didn't read my comment or forgot it :)

pH-7 replied:
I again @laiqujan I think you didn't read my comment or forgot it :)

pH-7 commented on src/simplejavatexteditor/UI.java:
```diff
@@ -596,5 +589,31 @@ private void saveFile() {
             }
         }
     }
+    private void openFile(){
```
Nice addition. I like it 😃

laiqujan committed:
Merge branch 'master' into master

laiqujan commented:
Thank-you @pH-7  for your kind response i resolved the conflict, let me know if any thing else creates problem.

pH-7 commented:
Great! Yes, there are also my above comments https://github.com/pH-7/Simple-Java-Text-Editor/pull/27#pullrequestreview-166680145 (check it directly on Github, it will be easier)

laiqujan committed:
#27 Changes Requested done.

https://github.com/pH-7/Simple-Java-Text-Editor/pull/27#pullrequestreview-166680145

laiqujan committed:
Changes requested have been done..

Changes requested have been done..

pH-7 commented on .gitignore:
```diff
@@ -2,8 +2,3 @@
-# Mac OS System
-.DS_Store*
-._*
-/bin/
```
Oh, I was only speaking about the one you added (`/bin/`). Not the ones above.

pH-7 commented:
@laiqujan Closing it until you fixed the mentioned issues. Please reopen it once done.
